src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -650,7 +650,12 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
 
                 string authority;
 
-                if (mergedOptions.IsB2C)
+                if (mergedOptions.PreserveAuthority && !string.IsNullOrEmpty(mergedOptions.Authority))
+                {
+                    authority = mergedOptions.Authority!;
+                    builder.WithGenericAuthority(authority);
```

Comment:
This will be switched to `WithOidcAuthority` right? Im assuming this is just for testing?

Reply:
`WithGenericAuthority` works today with MSAL 4.59. We soft deprecated this in MSAL 4.60, but this is yet to be released. In any case, it's just a rename. 

What doesn't work with MSAL 4.59 is an authority with extra query parameters, e.g. "https://cat.com/1235/v2.0?test=foo", but this is no longer needed (it used to be, the CIAM CUD authorities were behind a feature flag).